# Ecommerce Purchase

This dataset can be found on kaggle.com

In [11]:
# Importing the necessary libraries

import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt

In [5]:
df = pd.read_csv(r"C:\Users\Henry Oguns\Desktop\DATA ANALYSIS\Portfolio Projects\PortfolioProjects\Project _Datasets\Ecommerce Purchases")

In [3]:
# Display top 10 Rows of dataset 

df.head(10)

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                   Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3              7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4  23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   
5  7502 Powell Mission Apt. 768\nTravisland, VA 3...  21 XT       PM   
6     93971 Conway Causeway\nAndersonburgh, AZ 75107  96 Xt       AM   
7  260 Rachel Plains Suite 366\nCastroberg, WV 24...  96 pG       PM   
8            2129 Dylan Burg\nNew Michelle, ME 28650  45 JN       PM   
9    3795 Dawson Extensions\nLake Tinafort, ID 88739  15 Ug       AM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4  Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   
5  Mozilla/5.0 (Macintosh; U; PPC Mac OS X 10_8_5...   
6  Mozilla/5.0 (compatible; MSIE 7.0; Windows NT ...   
7  Mozilla/5.0 (X11; Linux i686) AppleWebKit/5350...   
8  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9  Mozilla/5.0 (X11; Linux i686; rv:1.9.7.20) Gec...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   
2       Simpson, Williams and Pham      675957666125       08/19   
3  Williams, Marshall and Buchanan  6011578504430710       02/24   
4        Brown, Watson and Andrews  6011456623207998       10/25   
5                   Silva-Anderson    30246185196287       07/25   
6                  Gibson and Sons  6011398782655569       07/24   
7                 Marshall-Collins      561252141909       06/25   
8                Galloway and Sons   180041795790001       04/24   
9     Rivera, Buchanan and Ramirez     4396283918371       01/17   

   CC Security Code                  CC Provider  \
0               900                 JCB 16 digit   
1               561                   Mastercard   
2               699                 JCB 16 digit   
3               384                     Discover   
4               678  Diners Club / Carte Blanche   
5              7169                     Discover   
6               714                VISA 16 digit   
7               256                VISA 13 digit   
8               899                 JCB 16 digit   
9               931             American Express   

                            Email                                     Job  \
0               pdunlap@yahoo.com  Scientist, product/process development   
1              anthony41@reed.com                       Drilling engineer   
2  amymiller@morales-harrison.com                Customer service manager   
3     brent16@olson-robinson.info                       Drilling engineer   
4     christopherwright@gmail.com                             Fine artist   
5               ynguyen@gmail.com                       Fish farm manager   
6              olivia04@yahoo.com                                  Dancer   
7            phillip48@parks.info                         Event organiser   
8            kdavis@rasmussen.com                       Financial manager   
9        qcoleman@hunt-huerta.com                      Forensic scientist   

        IP Address Language  Purchase Price  
0  149.146.147.205       el           98.14  
1     15.160.41.51       fr           70.73  
2   132.207.160.22       de            0.95  
3     30.250.74.19       es           78.04  
4     24.140.33.94       es           77.82  
5    55.96.152.147       ru           25.15  
6   127.252.144.18       de           88.56  
7   224.247.97.150       pt           44.25  
8  146.234.201.229       ru           59.54  
9    

In [18]:
df.shape

(10000, 14)

In [7]:
# checking information about the dataset to know the dtypes, memory used and if there is any missing values.

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


In [11]:
# check null values in the dataset and get the sum of the null value

df.isnull().sum()

Address             0
Lot                 0
AM or PM            0
Browser Info        0
Company             0
Credit Card         0
CC Exp Date         0
CC Security Code    0
CC Provider         0
Email               0
Job                 0
IP Address          0
Language            0
Purchase Price      0
dtype: int64

Showing from the above, the dataset lack any missing values

# Exploratory Data Analysis

In [19]:
# 1. what the highest and lowest purchase prices

df.columns

Index(['Address', 'Lot', 'AM or PM', 'Browser Info', 'Company', 'Credit Card',
       'CC Exp Date', 'CC Security Code', 'CC Provider', 'Email', 'Job',
       'IP Address', 'Language', 'Purchase Price'],
      dtype='object')

In [20]:
df['Purchase Price'].max()

99.99

In [21]:
df['Purchase Price'].min()

0.0

99.99 and 0 rupees are the highest and lowest purchase prices

In [23]:
#2. whats the Average purchase price

df['Purchase Price'].mean()

50.34730200000025

The average price in the dataset is Approximately 50.35 rupees

In [8]:
# 3. whats the number people that have the language french

df[df['Language'] == 'fr']

Address    Lot AM or PM  \
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
19                  125 Hall Summit\nBoothton, IL 41721  99 CU       PM   
53                PSC 9431, Box 7059\nAPO AA 29285-1363  14 qD       AM   
76    49206 Campbell Port\nNorth Cliffordshire, HI 3...  71 iu       PM   
82    493 Smith Valleys Suite 004\nNew Madelineville...  35 ls       PM   
...                                                 ...    ...      ...   
9941           43757 Brown Lodge\nValerieberg, PR 88518  68 DG       AM   
9947      32827 Carl Inlet\nSouth Nicole, NY 38081-6636  40 Dp       PM   
9951                518 Brown Dam\nGarciaside, IN 33057  71 wq       AM   
9977  02182 Keith Expressway\nEast Shannon, CT 20578...  34 RL       AM   
9980           6061 Dave Lights\nPhillipsview, UT 40500  25 Mv       AM   

                                           Browser Info  \
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
19    Mozilla/5.0 (compatible; MSIE 7.0; Windows NT ...   
53    Opera/9.34.(X11; Linux x86_64; it-IT) Presto/2...   
76    Mozilla/5.0 (Macintosh; U; PPC Mac OS X 10_5_9...   
82    Mozilla/5.0 (iPod; U; CPU iPhone OS 4_0 like M...   
...                                                 ...   
9941  Mozilla/5.0 (compatible; MSIE 5.0; Windows NT ...   
9947  Mozilla/5.0 (Windows; U; Windows 95) AppleWebK...   
9951  Mozilla/5.0 (compatible; MSIE 8.0; Windows NT ...   
9977  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9980  Opera/8.58.(Windows NT 5.0; it-IT) Presto/2.9....   

                              Company       Credit Card CC Exp Date  \
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
19                    Turner-Mckinney      676343504830       02/20   
53      Higgins, Cardenas and Kennedy   869972604798355       08/17   
76                      Jacobs-Tucker  6011343518820988       01/17   
82      Robinson, Johnston and Valdez  4351359627548412       06/17   
...                               ...               ...         ...   
9941         Jones, Williams and Dunn  6011508474487291       04/26   
9947                    Pruitt-Flores  3158564208754951       03/17   
9951      Crosby, Vasquez and Ballard  5256758441931287       12/17   
9977      Deleon, Jacobson and Benton  4186094003664688       06/21   
9980                      Cortez-Frye  6011944199756993       12/24   

      CC Security Code    CC Provider                     Email  \
1                  561     Mastercard        anthony41@reed.com   
19                 440  VISA 16 digit      ruiznicole@gmail.com   
53                 157   JCB 16 digit        amorales@yahoo.com   
76                 806        Voyager        eperez@hotmail.com   
82                 937  VISA 13 digit         amendez@yahoo.com   
...                ...            ...                       ...   
9941               906     Mastercard     christian55@gmail.com   
9947               386     Mastercard     josephsmith@gmail.com   
9951               146  VISA 16 digit       cjennings@yahoo.com   
9977               397   JCB 16 digit    daltoncarter@yahoo.com   
9980               755   JCB 15 digit  mark74@kirby-stewart.org   

                                               Job       IP Address Language  \
1                                Drilling engineer     15.160.41.51       fr   
19                      Designer, interior/spatial   25.105.209.214       fr   
53                                Technical author   44.108.117.122       fr   
76                                       Paramedic   210.207.58.168       fr   
82                             Engineer, materials  163.129.163.100       fr   
...                                            ...              ...      ...   
9941                      Secondary school teacher  216.118.235.145       fr   
9947                              Textile designer     198.68.38.23       fr   
9951  Lighting technician, broadcasting/film/video       58.

In [9]:
len(df[df['Language'] == 'fr'])

1097

We have a total of 1,097 people with the language french

In [35]:
# 4. Number of job title that contains Enginer

df['Job']

0       Scientist, product/process development
1                            Drilling engineer
2                     Customer service manager
3                            Drilling engineer
4                                  Fine artist
                         ...                  
9995                                Printmaker
9996                           Energy engineer
9997                        Veterinary surgeon
9998                  Local government officer
9999                    Embryologist, clinical
Name: Job, Length: 10000, dtype: object

In [40]:
len(df[df['Job'].str.contains('engineer', case=False)])

984

There 984 people with the job title that contains engineer

In [47]:
# 5. Find the email of the person with the ip address: 132.207.160.22

df[df['IP Address'] =='132.207.160.22']

Address    Lot AM or PM  \
2  Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   

                                        Browser Info  \
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   

                      Company   Credit Card CC Exp Date  CC Security Code  \
2  Simpson, Williams and Pham  675957666125       08/19               699   

    CC Provider                           Email                       Job  \
2  JCB 16 digit  amymiller@morales-harrison.com  Customer service manager   

       IP Address Language  Purchase Price  
2  132.207.160.22       de            0.95

In [48]:
df[df['IP Address'] =='132.207.160.22']['Email']

2    amymiller@morales-harrison.com
Name: Email, dtype: object

 amymiller@morales-harrison.com is the email address fr the person with the ip address: 132.207.160.22

In [52]:
# 6. how many people have mastercard as their credit card provider and made a purchase above 50

df[(df['CC Provider'] =='Mastercard') & (df['Purchase Price'] > 50)]

Address    Lot AM or PM  \
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
18    461 Christopher Square\nWest Michaelchester, C...  17 SB       PM   
31                     USNS Alvarado\nFPO AA 27052-1231  26 Lh       PM   
35    93392 Webb Gardens Apt. 220\nLaurabury, AR 999...  37 om       AM   
90          431 Bowen Lights\nFergusonborough, MH 01362  31 tG       AM   
...                                                 ...    ...      ...   
9941           43757 Brown Lodge\nValerieberg, PR 88518  68 DG       AM   
9948  95544 Johnson Isle Suite 939\nMichaelberg, RI ...  91 bW       AM   
9954  051 Samantha Wells Apt. 328\nScottborough, OR ...  11 vG       AM   
9981  087 Larson Roads Apt. 587\nRicestad, WI 98077-...  95 Kw       AM   
9987     2754 Klein Mission\nJohnsonview, PA 64925-9624  65 Fg       PM   

                                           Browser Info  \
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
18    Mozilla/5.0 (X11; Linux i686; rv:1.9.6.20) Gec...   
31    Opera/8.84.(X11; Linux i686; sl-SI) Presto/2.9...   
35    Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
90    Mozilla/5.0 (Windows NT 5.2; it-IT; rv:1.9.1.2...   
...                                                 ...   
9941  Mozilla/5.0 (compatible; MSIE 5.0; Windows NT ...   
9948  Opera/8.36.(X11; Linux x86_64; sl-SI) Presto/2...   
9954  Mozilla/5.0 (Macintosh; U; PPC Mac OS X 10_8_2...   
9981  Opera/9.63.(Windows NT 5.01; en-US) Presto/2.9...   
9987  Opera/9.26.(Windows CE; sl-SI) Presto/2.9.188 ...   

                              Company       Credit Card CC Exp Date  \
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
18             Beard, Abbott and Pena  6011350184276270       12/22   
31                    Nicholson Group     4614997834548       03/22   
35                           Mora Ltd  6011049630969815       09/16   
90                       Copeland-Lee      639032576097       06/18   
...                               ...               ...         ...   
9941         Jones, Williams and Dunn  6011508474487291       04/26   
9948                     Fox-Peterson     4762924304307       03/17   
9954    Mendoza, Zimmerman and Reilly    30232295374907       05/21   
9981         Ortiz, Schultz and Adams     4690863684605       01/19   
9987                       Curtis LLC   180027305702263       12/24   

      CC Security Code CC Provider                       Email  \
1                  561  Mastercard          anthony41@reed.com   
18                 767  Mastercard          hannah63@yahoo.com   
31                 909  Mastercard        ashley12@hotmail.com   
35                 367  Mastercard      hgonzalez@mcdowell.com   
90                 868  Mastercard           nrogers@brown.com   
...                ...         ...                         ...   
9941               906  Mastercard       christian55@gmail.com   
9948               567  Mastercard     haleybenjamin@gmail.com   
9954               762  Mastercard        brownamy@perkins.com   
9981               695  Mastercard  laurenbennett@richards.com   
9987               865  Mastercard    denisehamilton@novak.biz   

                                         Job       IP Address Language  \
1                          Drilling engineer     15.160.41.51       fr   
18                              Photographer   73.250.176.201       el   
31                           Sales executive   94.176.142.201       en   
35               Lecturer, further education     216.23.95.40       zh   
90    Teacher, English as a foreign language   153.188.13.203       es   
...                                      ...              ...      ...   
9941                Secondary school teacher  216.118.235.145       fr   
9948                     Structural engineer    120.36.140.58       en   
9954                 Presenter, broadcasting     61.97.84.101       it   
9981         Environmental education officer    106.

In [53]:
len(df[(df['CC Provider'] =='Mastercard') & (df['Purchase Price'] > 50)])

405

We have 405 people that have mastercard as their credit card provider and have made a purchase above 50

In [65]:
# 7. ind the email of the person with the following credit card number: 4664825258997302

df.head(2)

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   

   CC Security Code   CC Provider               Email  \
0               900  JCB 16 digit   pdunlap@yahoo.com   
1               561    Mastercard  anthony41@reed.com   

                                      Job       IP Address Language  \
0  Scientist, product/process development  149.146.147.205       el   
1                       Drilling engineer     15.160.41.51       fr   

   Purchase Price  
0           98.14  
1           70.73

In [69]:
df[df['Credit Card'] ==4664825258997302]['Email']

9992    bberry@wright.net
Name: Email, dtype: object

bberry@wright.net is the email of the person with the credit card number: 4664825258997302

In [71]:
# 8. how many people purchase during the m and how many people purchased during pm

df['AM or PM'].value_counts()

PM    5068
AM    4932
Name: AM or PM, dtype: int64

We have 5,068 and 4,932 people purchasing during pm and am

In [75]:
# 9. how many people have a credit card that expires in 2020

df['CC Exp Date']

0       02/20
1       11/18
2       08/19
3       02/24
4       10/25
        ...  
9995    03/22
9996    07/25
9997    05/21
9998    11/17
9999    02/19
Name: CC Exp Date, Length: 10000, dtype: object

In [79]:
def fun():
    count = 0
    for date in df['CC Exp Date']:
        if date.split('/')[1] =='20':
            count=count+1
    print(count)
    

In [80]:
fun()

988


We 988 people having their credit card expire in 2020

In [84]:
# 10. Top 5 most popular email providers (eg. gmail.com etc)

df['Email']

0                    pdunlap@yahoo.com
1                   anthony41@reed.com
2       amymiller@morales-harrison.com
3          brent16@olson-robinson.info
4          christopherwright@gmail.com
                     ...              
9995            iscott@wade-garner.com
9996                mary85@hotmail.com
9997                 tyler16@gmail.com
9998           elizabethmoore@reid.net
9999             rachelford@vaughn.com
Name: Email, Length: 10000, dtype: object

In [85]:
list1 = []
for email in df['Email']:
    list1.append(email.split('@')[1])

In [86]:
df['temp'] = list1

In [87]:
df.head()

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                   Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3              7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4  23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4  Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   
2       Simpson, Williams and Pham      675957666125       08/19   
3  Williams, Marshall and Buchanan  6011578504430710       02/24   
4        Brown, Watson and Andrews  6011456623207998       10/25   

   CC Security Code                  CC Provider  \
0               900                 JCB 16 digit   
1               561                   Mastercard   
2               699                 JCB 16 digit   
3               384                     Discover   
4               678  Diners Club / Carte Blanche   

                            Email                                     Job  \
0               pdunlap@yahoo.com  Scientist, product/process development   
1              anthony41@reed.com                       Drilling engineer   
2  amymiller@morales-harrison.com                Customer service manager   
3     brent16@olson-robinson.info                       Drilling engineer   
4     christopherwright@gmail.com                             Fine artist   

        IP Address Language  Purchase Price                  temp  
0  149.146.147.205       el           98.14             yahoo.com  
1     15.160.41.51       fr           70.73              reed.com  
2   132.207.160.22       de            0.95  morales-harrison.com  
3     30.250.74.19       es           78.04   olson-robinson.info  
4     24.140.33.94       es           77.82             gmail.com

In [88]:
df['temp'].value_counts()

hotmail.com              1638
yahoo.com                1616
gmail.com                1605
smith.com                  42
williams.com               37
                         ... 
callahan-robinson.com       1
brown-mckay.com             1
benson.net                  1
walter-scott.net            1
wilkins-campos.com          1
Name: temp, Length: 3416, dtype: int64

In [89]:
df['Email'].apply(lambda x:x.split('@')[1]).value_counts().head()

hotmail.com     1638
yahoo.com       1616
gmail.com       1605
smith.com         42
williams.com      37
Name: Email, dtype: int64

The top 5 most email providers are Hotmail.com, Yahoo.com, gmail.com, smith.com and williams.com 